src/argprocessing.cpp

```diff
@@ -1116,7 +1116,19 @@ process_args(Context& ctx)
   }
 
   // Determine output object file.
-  const bool implicit_output_obj = args_info.output_obj.empty();
+  bool implicit_output_obj = args_info.output_obj.empty();
+  if (!implicit_output_obj && ctx.config.is_compiler_group_msvc()) {
+    if (*args_info.output_obj.rbegin() != '\\') {
+      implicit_output_obj = true;
```

Comment:
I don't understand this logic. If `args_info.output_obj` is not empty, then by definition it's not an implicit output object since an output object filename was passed explicitly.

Also, doesn't this break `/Fo` with a filename? If the compiler is MSVC and `example.c` and `/Foexample.obj` is on the command line, then `!implicit_output_obj && ctx.config.is_compiler_group_msvc()` will be true. Since `*args_info.output_obj.rbegin() != '\\'` will be true, `implicit_output_obj` will be set to true. On line 1132, `implicit_output_obj && !args_info.input_file.empty()` will be true, so `args_info.output_obj` will be set to `example.objexample.obj`. Am I missing something?

Reply:
Oops, this should have been == of course!

I reused `implicit_output_obj` to avoid duplication of the extension translation logic. Do you prefer a separate variable for clearance? It will be equivalent to `implicit_output_obj` though.

Reply:
I would prefer a  a more descriptive name of `implicit_output_obj`, or separate variable if that makes the code more readable.

Reply:
Done